crates/biome_js_analyze/src/lint/nursery/no_array_constructor.rs

```diff
@@ -0,0 +1,99 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, Ast, FixKind, Rule, RuleDiagnostic, RuleSource,
+};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsCallArgument, AnyJsExpression, JsCallArguments, JsExpressionStatement};
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow Array constructors.
+    ///
+    /// Use of the Array constructor to construct a new array is generally discouraged in favor of array literal notation because of the single-argument pitfall and because the Array global may be redefined.
+    /// The exception is when the Array constructor is used to intentionally create sparse arrays of a specified size by giving the constructor a single numeric argument.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// Array(0, 1, 1);
+    /// ```
+    ///
```

Comment:
Please add more invalid cases as ESLint does. These help users understand the rules.

```suggestion
    /// ```js,expect_diagnostic
    /// Array();
    /// ```
    ///
    /// ```js,expect_diagnostic
    /// Array(0, 1, 2);
    /// ```
    ///
    /// ```js,expect_diagnostic
    /// new Array(0, 1, 2);
    /// ```
    ///
    /// ```js,expect_diagnostic
    /// Array(...args);
    /// ```
    ///
```